### ISSUE TITLE
Update SSL UI in Dashboard and doc

### ISSUE THREAD

**Issue by nataliejeremy** [pantheon-systems/documentation#818]
First off, we already had SSL certs so it was a matter of having them rekeyed (simple enough) and uploading them to Pantheon.  Didn't use CloudFlares.

Exec summary is the interface is simple (awesome), but the descriptions/documentations have some holes that make it frustrating, even for a technical user.  The good news is that I think you guys can do some really simple things with descriptions and documentation to make it more obvious what goes where (I ended up probably spending 10-15 minutes bouncing between help pages and googling terms just to know how to fill in these three boxes correctly):

![ssl](https://cloud.githubusercontent.com/assets/9138609/9309095/610bbf14-44bc-11e5-9c78-8e883640ed7e.png)

Here's a blow-by-blow description of my thought process, just to give you the full effect :) ...

To start with, the top box is clear ("Certificate"..."paste the .crt file here").  Awesome!

The second box says "Intermediary Certificate".  What is that?  (I'm pretty darn technical and I've deployed a ton of SSL certs before, but I've never heard this term).  The description says "if you have one, put it here."  That doesn't really help...

Maybe the help doc will clarify.  Hmm...the it just says "Paste in the requested information and press Add Cert."  That doesn't help either...

If I scroll down further, there's a section on "Intermediate Certificate"...bingo!  Wait, it still doesn't tell me what exactly it is:

"Your SSL provider may give you an intermediary certificate, which is used establish a chain of trust. If you get more than one intermediary certificate file, you'll need to combine them into one. Your provider may also send you a "certificate authority" or "CA" cert. If you have problems with your intermediate certificate, try https://whatsmychaincert.com/, which tests if the correct certificate chain is being served, and tells you which chain you should be serving."

None of this actually tells me what an intermediate certificate file is, it just says what it does - so I'm really no closer to knowing whether or not I have one (FWIW, I tried clicking on the link and I have no clue what the point of that website is, but it certainly didn't help).  Well...I guess I don't have one, so I'll leave it blank (turns out I did...more on that later).

The third box says "RSA Private Key", but again no description of what that is.  I had to Google to know that the RSA Private Key was the .key file (someone super into security would probably know that, but for an average user who does something with SSL certs once a year and hasn't memorized the terms, this one could easily be solved by having a label like the top box "paste the .key file here").

Ok, so I've got a certificate and a private key in there now, and I press "Add Cert."

...and I get an error message saying that my certificate was invalid.  (So much for the first box being easy).

I triple-checked to make sure I had the right certificate, copied and pasted a couple times to make sure I didn't screw up, but still no luck.  After flailing for a few minutes, I finally figured out that the validation algorithm requires the certificate text to include the "-----BEGIN CERTIFICATE REQUEST-----" header and "-----END CERTIFICATE REQUEST-----" footers (something other sites I've used in the past don't require).  In other words, this worked:

-----BEGIN CERTIFICATE REQUEST-----
x
y
z
-----END CERTIFICATE REQUEST-----

but this did not:

x
y
z

Two obvious ways to fix: (1) engineering: if the header/footer isn't present, try adding it and re-test, or (2) documentation: add a note (or a tooltip) pointing out that you need to include the header and footer.  Done!

...ok, so we've got the certificate ready.  Now....go!

Or not.

I got another error message (which, unfortunately, I can't remember the exact details of) which said something along the lines of "the certificate is incomplete."  Reasonable leap of faith: I guess I do need an "Intermediate Certificate."

Too bad I still don't know what the heck it is or where I find it.

Wait...GoDaddy includes a "certificate bundle" with the certificate.  I've never before used it before as a separate file (e.g in apache or nginx you just drop both files in), but maybe that's it...?

Yep, turns out what GoDaddy calls a bundle is the Intermediate Certificate.  Drop it in there, and success!

My next thought: I can't possibly be the only GoDaddy customer who's ran into this...let me look (for the fourth time) at the documentation. I went back yet again to the help doc, scrolled all the way to the bottom and found an "FAQ" section with this as the first question:
What should I do with a GoDaddy gd_bundle.crt?

Use it as the Intermediary Certificate.

Man...that would have been super helpful to have had in the description below the second box of the SSL form.  Or as a tooltip beside the word "Intermediate Certificate."  Or in the description on the help page under the section titled "Intermediate Certificate."  Or any other place I looked in the 10 minutes I spent trying to figure out whether or not I had one :)

In summary, it could have been a super-simple 30 second process, if only the documentation was a little bit better :).  Other than that, no complaints!

Submitted by Zack from customer feedback. See email sent on Aug 16, 6:17 PM

**Comment by ari-gold** [comment]
Will work be done to help text for Dashboard? Where is that being tracked? 

Suggest adding something like following help text to "Intermediate Certificate": 

You can use https://whatsmychaincert.com/ to generate your correct intermediate certificate.

**Comment by nataliejeremy** [comment]
@ari-gold Shawn and Nathan are working on the UI changes. We aren't tracking those here (probably in  sprintly).
